tests/integration/leases_test.go

```diff
@@ -14,134 +14,140 @@ import (
 	"time"
 )
 
+type cliInputs []string
+
+type cliOutputs struct {
+	httpRequest interface{}
+	printedJson map[string]interface{}
+	printedString string
+}
+
+
 func TestLeasesCreate(t *testing.T) {
 
-	t.Run("with --expires-on flag", func(t *testing.T) {
```

Comment:
@joshmarsh I was intentionally avoiding table-driven tests here. They make it impossible to run a single test case at a time, and make IDE integration clunky. 

If our goal is to avoid code duplication between test, we get the same result from the kind of abstraction like you're seeing here, eg:

```
t.Run("should add positive numbers", func(t *testing.T) {
  addTest(t, &addTestCase{
    input: []int{5, 6},
    expectedOutput: 11
  })
})


t.Run("should add negative numbers", func(t *testing.T) {
  addTest(t, &addTestCase{
    input: []int{-5, -6},
    expectedOutput: -11
  })
})

func addTest(t *testing.T, case *addTestCase) {
  res := add(case.input)
  assert.Equal(t, case.expectedOutput, res)
}

Reply:
I think table driven tests make sense here because CLI commands can typically be executed in a number of different ways given different permutations of args and flags, and grouping those test cases together into a “table” makes it easier to understand what permutations are being tested and what aren’t. I also think it makes it easier to reason about your test inputs and outputs. Table driven tests are a pretty standard idiom in go. e.g. https://github.com/golang/go/blob/master/src/cmd/api/goapi_test.go#L101. I prefer to stick to common idioms as it makes the code more readable for people that are used to seeing the same patterns used elsewhere.

I haven’t had the need to execute an individual test case from the terminal before. What’s your use case for doing so?

Reply:
Table driven tests are also pretty widely used throughout DCE itself. Are we moving away from that pattern entirely?